src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs

```diff
@@ -99,14 +93,22 @@ private void ConnectionStop(string connectionId)
         [NonEvent]
         public void RequestStart(HttpProtocol httpProtocol)
         {
-            // avoid allocating the trace identifier unless logging is enabled
             if (IsEnabled())
             {
-                RequestStart(httpProtocol.ConnectionIdFeature, httpProtocol.TraceIdentifier, httpProtocol.HttpVersion, httpProtocol.Path!, httpProtocol.MethodText);
+                Core(httpProtocol);
+            }
+
+            [NonEvent]
+            void Core(HttpProtocol httpProtocol)
+            {
+                // avoid allocating the trace identifier unless logging is enabled
+                if (IsEnabled(EventLevel.Informational, EventKeywords.None))
+                {
+                    RequestStart(httpProtocol.ConnectionIdFeature, httpProtocol.TraceIdentifier, httpProtocol.HttpVersion, httpProtocol.Path!, httpProtocol.MethodText);
+                }
             }
```

Comment:
Seeing that `MethodImplOptions.NoInlining` was explicitly used here, I'm assuming you were aiming at reducing the size of the main request loop? I can remove this extra method if you don't think it's needed.

Reply:
I'd leave it for now.